backend/tests/v5/e2e/routes/sso/aad/aad.test.js

```diff
@@ -254,6 +254,181 @@ const signupPost = () => {
 	});
 };
 
+const testLink = () => {
+	describe('Link', () => {
+		const redirectUri = generateRandomURL();
+
+		test('should fail without a valid session', async () => {
+			await agent.get(`/v5/sso/aad/link?redirectUri=${redirectUri}&key=${testUser.apiKey}`)
+				.expect(templates.notAuthorized.status);
+		});
+
+		describe('With valid authentication', () => {
+			beforeAll(async () => {
+				await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			});
+			afterAll(async () => {
+				await testSession.post('/v5/logout/');
+			});
+
+			test('should fail without a valid state', async () => {
+				const state = generateRandomString();
+				const res = await testSession.get(`/v5/sso/aad/link-post?state=${state}`)
+					.expect(templates.invalidArguments.status);
+				expect(res.body.code).toEqual(templates.invalidArguments.code);
+			});
+
+			test('should fail if redirectUri is not provided', async () => {
+				await testSession.get('/v5/sso/aad/link')
+					.expect(templates.invalidArguments.status);
+			});
+
+			test('should redirect the user to Microsoft authentication page', async () => {
+				const res = await testSession.get(`/v5/sso/aad/link?redirectUri=${redirectUri}`)
+					.expect(302);
+				const resUri = new URL(res.headers.location);
+				expect(resUri.hostname).toEqual('login.microsoftonline.com');
```

Comment:
I think you can test both at the same time with `href`
```suggestion
				expect(resUri.href).toEqual('https://login.microsoftonline.com/blah...');
```

Reply:
href is this whole thing 

![image](https://user-images.githubusercontent.com/71635617/207111336-20361836-e9a6-498d-a109-2a0eda035305.png)